### ISSUE TITLE
Convert to JPEG XL with splines

### ISSUE THREAD

**Issue by niutech** [yisibl/resvg-js#111]
JPEG XL is a successor to JPG and PNG formats and it allows to store vector paths using centripetal Catmull-Rom splines, which are much more compact than raw pixel data and can be upscaled to any resolution preserving sharpness. You can generate such JXL files programmatically by converting SVG paths to splines, see the [JXL-Art example](https://jxl-art.surma.technology/?zcode=7Zbfiuc1DMXv-xR9gZWmSf_dugheu4rXwg7uwLoMOO-Pn9N2XcHLWXEQGfidab9JkyYnSb99fH7_8PT8IUf66en3X357-viQa_r58T1bvZT0_cPjrx-es_PvD29_zCW9e_r4-OkhvbGWy0v-XsUJ5Zv60iNaydZKIkAZ-O7T-xugrxan8q9Hqb48Sha5FkvWPmO72MF60Q_GuDiPfCzQD_a7PBBHKI5O7KP-loHyf_wPPc1WqrNmm_WiXwzQDpqDmLR60f6ytlT7ytV6qqOA4-K8yPlBhgbrIEVD67jYD66Zq_N9rY1eDBwXWbcj7yLImBfX3Wftx87Bz2v03bZfPuf28yD3nuviXdey7-m2Ls4dh4P1zzh9dRa9hm73FXgUE36QpxBvooPwo0WKQTy7gQNcIHKD7_DCyF_0sfMSPXa8o588BIV-8PAwfOw8hPvBuo48ntvkvMr-wo-K_IqL-FPJf-l7XU37sXkVFd41nTNy7WWfVzvyjtxo4OGV7Fb5FzQa2YtxUPyTH1376A_4q3st6an72z6vFfZ7S02tiTg0-FQb-_CtNvblV7ODoTV6UZHDjkseP-s463r14fPWNwfVIPkuv-lyzde-d6OuFIe247hS6yeOrc8d97bjzzmD_YHeIP7ydxDPHgflp_aZX1tOfknf8VP5qnxvrCv7yhP1tu3a9aPM7ZcVzlOeyrmXFcVnXdQ-8V1aE9eFvvK1fOvHgkfGd9WpkZfZtr2Y2PcvvHuF87W8gum63x76bYlG3fZo7FXjstOrax507pnhgbcsFnuG27xcqCgSO2fqPVNVkGBZorRXTavs35kXY5nsCRgO7Bnc3bCkY1TOlMHIU4O75okTlPeU_ZEhoXEutUb_zZSMkVqYZ3NkCgXS5oYe5ppv0GAXSGHmkEjPpB5aZ0dPFagzVUhY0EMCs9BN1mE7mwEZsU5tG98oHeQpKM4gRvwWIpJW4wcnZqJ38WMSpsxhoYoJjkJFmKhGxXyhDNNuJomSdJnM3rGRXYfnUJSyniH_1DvwPzJBnFS1C8zftkRUp20xNRwaiqTkVrOYTEMphy_wzNcmn9NAoGKU2ADbYWwwzefczX7PENJ7WjodL0ggjTngBDxTf2--gZejNkMiJoKFz0wr3iNBE0QvgjNgZEj9C7PiS9D08YsWNvbDAsc1BuWxuivXgKK6lGYGAF8aN50iyb3-F368ye8eniHaHw).
Could you implement conversion from SVG to JPEG XL using splines?

**Comment by yisibl** [comment]
Thanks for the suggestion, `Catmull-Rom splines` was very interesting. I found a related article [SVG to JPEG XL](https://grass.moe/splines)

Further research revealed that the biggest obstacle to converting SVG to JXL is the blurring when zoomed in. This is because the current decoder (libjxl) does not sharpen by default, depending on the https://github.com/libjxl/libjxl/issues/1370#issuecomment-1106406131

So we need to wait for these infrastructures to mature before we consider them.

**Comment by yisibl** [comment]
Additions:

Splines there is no notion of 'fill', we can only do curves with them, not filled surfaces. But we could "emulate" filled shapes by using thick strokes.

Did I understand correctly?

**Comment by niutech** [comment]
Yes, this is what I am working on currently - surface fill SVG shapes using thick lines. As for sharpen lines, you can generate upsampled (e.g. 4x) JXL images having the same file size as 1x, then resize them back to 1x size using CSS attributes: `width`, `height` and `image-rendering: -webkit-optimize-contrast` or `crisp-edges`.
Compare 1000px JXL example image downscaled to 250px (left) with SVG at the same dimensions (right): 
![example](https://user-images.githubusercontent.com/384997/168444606-2620f0c1-7f1f-40be-a2e9-dd90a0231b45.png)

**Comment by TPS** [comment]
[svg2jxl](https://github.com/meiMingle/spline) might be useful?
